crates/crypto/src/signing/private_key.rs

```diff
@@ -17,6 +17,12 @@ pub enum PrivateKeyInner {
 }
 
 impl PrivateKey {
+    /// Read a private key from a file.
+    pub fn from_file(path: impl AsRef<Path>) -> Result<Self, PrivateKeyParseError> {
```

Comment:
Is there a reason we need to add a `from_file` here and adding IO to the error set?
It feels like we're mixing abstractions and you could just as easily `read_to_string(...).parse()` to keep them separate.

Reply:
My main objection is the `PrivateKeyParseError::IO` addition which isn't actually caused by parsing or reachable when calling `parse`.

Reply:
I stuck it in here because of `Zeroizing`, but I could probably shuffle things around to avoid it.

Reply:
Just curious bould this just return an `anyhow::Result` so as to not pollute `PrivateKeyParseError`? I find `from_file` conventions to be rather useful for public items like this but fine either way.

Reply:
Switched `FromStr` -> `TryFrom<String>` which lets it zeroize.

Reply:
Well you can [see the diff here](https://github.com/bytecodealliance/registry/compare/ed730919cf3f77f95e9e583963916cd8f0cb07c7..8d68811780d8e1b8f639ee8d919c5e4f8b875a62), but github seems to be having trouble with consistency... :grimacing:

Reply:
Had to push an empty diff to convince GH things had actually changed :roll_eyes: .